### PR TITLE
[v1.x] Add onnx export support for where and greater_scalar operators.

### PR DESCRIPTION
**Pull request by josephevans**
## Description ##
Add onnx export support and unit tests for the "where" and "greater_scalar" operators.

### CODE REVIEW

File: tests/python-pytest/onnx/test_operators.py [hunk 1]
```diff
@@ -378,3 +378,27 @@ def test_onnx_export_contrib_BilinearResize2D(tmp_path, dtype, params):
     x = mx.nd.arange(0, 160).reshape((2, 2, 5, 8))
     M = def_model('contrib.BilinearResize2D', **params)
     op_export_test('contrib_BilinearResize2D', M, [x], tmp_path)
+
+
+@pytest.mark.parametrize("dtype", ["float32", "float64", "int32", "int64"])
```
Comment (Zha0q1) [inline tests/python-pytest/onnx/test_operators.py:383]: shall we also try float16

**Review by Zha0q1 — APPROVED** [review]
LGTM

File: tests/python-pytest/onnx/test_operators.py (hunk 1 above)
Comment (josephevans) [reply tests/python-pytest/onnx/test_operators.py]: Update to work with float16 and also test it :) Thanks for the help.

**Review by waytrue17 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**